src/main/java/hudson/plugins/git/extensions/impl/SubmoduleOption.java

```diff
@@ -42,18 +42,20 @@
     private String reference;
     private boolean parentCredentials;
     private Integer timeout;
+    private Integer threads = 1;
     /** Use --depth flag on submodule update command - requires git>=1.8.4 */
     private boolean shallow;
     private int depth = 1;
 
     @DataBoundConstructor
-    public SubmoduleOption(boolean disableSubmodules, boolean recursiveSubmodules, boolean trackingSubmodules, String reference,Integer timeout, boolean parentCredentials) {
+    public SubmoduleOption(boolean disableSubmodules, boolean recursiveSubmodules, boolean trackingSubmodules, String reference, Integer timeout, boolean parentCredentials) {
```

Comment:
I tend to update primary constructors like the following:

```java
@DataBoundConstructor
public SubmoduleOption(boolean disableSubmodules, boolean recursiveSubmodules, boolean trackingSubmodules, String reference, Integer timeout, boolean parentCredentials, Integer threads) {
    // cool code goes here
}

//but we still need to migrate older configurations so...
@Deprecated
public SubmoduleOption(boolean disableSubmodules, boolean recursiveSubmodules, boolean trackingSubmodules, String reference,Integer timeout, boolean parentCredentials) {
    this(disableSubmodules, recursiveSubmodules, trackingSubmodules, reference, timeout, parentCredentials, 1);
    // don't forget the 1 is to set migrating configurations to 1 thread by default.
}
```

Reply:
Constructors should only be extended for mandatory parameters - see https://jenkins.io/doc/developer/plugin-development/pipeline-integration/#constructor-vs-setters. 
This also limits the need to deprecate old constructor versions.
Unfortunately this rule hasn't be followed in the past for many plugins.

Reply:
Interesting, thanks for the documentation link.  I was not aware before but am now.